docs/design/factories.md

```diff
@@ -432,6 +432,36 @@ however, this avenue will no longer be available, and we'll either need to
 surface the resource type or its schema in the CRUD functions somehow, or
 decide that provider developers don't need access to it.
 
+## Recommendations
+
+We believe the reflection and manually copying options are dangerous and are
+strongly recommending against them. They're not idiomatic and they hide too
+much of what's happening.
+
+Named functions and anonymous functions are mostly useful on structs, and we're
+shying away from that pattern, so we're recommending against them.
```

Comment:
Perhaps link to the structs/interfaces doc here

Reply:
Good call. :+1: